BlockSettleHW/ledger/ledgerDevice.cpp

```diff
@@ -1118,15 +1130,42 @@ void LedgerCommandThread::releaseDevice()
 bool LedgerCommandThread::exchangeData(const QByteArray& input,
    QByteArray& output, std::string&& logHeader)
 {
-   auto logHeaderCopy = logHeader;
-   if (!writeData(input, std::move(logHeader))) {
+   if (!writeData(input, logHeader)) {
       return false;
    }
 
-   return readData(output, std::move(logHeaderCopy));
+   try {
+      return readData(output, logHeader);
+   }
+   catch (std::exception &e){
+      // Special case : sometimes hidapi return incorrect sequence_index as response
+      // and there is really now good solution for it, except restart dongle session
+      // till the moment error gone.
+      // https://github.com/obsidiansystems/ledger-app-tezos/blob/191-troubleshooting/README.md#error-unexpected-sequence-number-expected-0-got-191-on-macos
+      // (solution in link above is not working, given here for reference)
+      // Also it's a general issue for OSX really, but let's left it for all system, just in case
+      // And let's have 10 times threshold to avoid trying infinitively
+      static int maxAttempts = 10;
+      if (e.what() == kHidapiSequence191 && maxAttempts > 0) {
+         --maxAttempts;
+         OnExitCb([&] {
```

Comment:
Nameless temporary variable will be destroyed right on the next line.

Reply:
true, will fix